### PR TITLE
2455 replace all boost pointers by std pointers (Part. 1)

### PR DESCRIPTION
**Pull request by barondim**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [x] the corresponding milestone was added in the ticket and in this PR
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated

### CODE REVIEW

File: dynawo/sources/Solvers/Common/DYNSolverFactory.h [hunk 1]
```diff
@@ -60,13 +61,15 @@ class SolverFactory {
    */
   virtual void destroy(Solver*) const = 0;
 
+  using SolverPtr = std::unique_ptr<Solver, std::function<void(Solver*)> >;
```
Comment (gautierbureau) [inline dynawo/sources/Solvers/Common/DYNSolverFactory.h:64]: second template argument is supposed to be a deleter I am not sure this is the case here

**Review by gautierbureau — CHANGES_REQUESTED** [review]

**Review by rosiereflo — CHANGES_REQUESTED** [review]
Did you make sure satellite and private projets are compiling? Some of those APIs might be used

File: dynawo/sources/Solvers/Common/DYNSolverFactory.h (hunk 1 above)
Comment (barondim) [reply dynawo/sources/Solvers/Common/DYNSolverFactory.h:64]: The deleted is added in dynawo/sources/Solvers/Common/DYNSolverFactory.cpp :

`solverPtr = SolverPtr(solver, deleteSolver);`

**Comment by barondim** [comment]
> Did you make sure satellite and private projets are compiling? Some of those APIs might be used

Yes, everything compiles

**Review by rosiereflo — APPROVED** [review]

**Comment by barondim** [comment]
- [ ] Do not merge

**Comment by barondim** [comment]
- [ ] Do not merge

**Review by gautierbureau — APPROVED** [review]

**Review by rosiereflo — APPROVED** [review]

**Review by gautierbureau — APPROVED** [review]

**Pull request merged**